### ISSUE TITLE
Nested factories for the operations plugin

### ISSUE THREAD

**Issue by ertrzyiks** [zhouzi/graphql-codegen-factories#31]
In [the example from readme](https://stackblitz.com/github/zhouzi/graphql-codegen-factories/tree/main/examples/basic?file=generated%2Ftypes.ts) I noticed that the main plugin is based on the nested factories. Factory for one type reuses the factory for another type.

```
export function createUserMock(props: Partial<User>): User {
  return {
    __typename: "User",
    companion: createDogMock({}),
    id: "",
    role: UserRole.Admin,
    ...props,
  };
}

```
The operation plugins however returns just a minimal object that matches the query response interface. It has limited value as to provide data for different scenarios we still need to write huge part of the responses by hand. The idea from the main plugin applied to the operations plugin would be a game changer in the graphql mocking in my opinion.

My current output:
```
export type LoadCategoriesQuery = {
  __typename?: "Query",
  allRecipeCategory: {
    __typename?: "RecipeCategoryConnection",
    edges: Array<{
      __typename?: "RecipeCategoryEdge",
      node: {
        __typename?: "RecipeCategory",
        name?: string | null,
        slug?: string | null,
      },
    }>,
  },
};


export function createLoadCategoriesQueryMock(
  props: Partial<LoadCategoriesQuery>
): LoadCategoriesQuery {
  return {
    allRecipeCategory: {
      __typename: "RecipeCategoryConnection",
      edges: [],
    },

    ...props,
  };
}
```

To provide a fake response with a single edge I still have plenty of code to write. It would be awesome to additionally have a few more factories for the collections:

```
export function createLoadCategoriesQueryRecipeCategoryEdgeMock(){  return ({ 
  __typename: 'RecipeCategoryEdge',
  node: {
    __typename: 'RecipeCategory',
    name: '',
    slug: ''
  }
}) }
```

**Comment by zhouzi** [comment]
Agreed, changing the default value of a nested property is a pain. We can't name the factories based on the type though as the following is possible:

```graphql
query GetRecipes {
  topRecipes: recipes(orderBy: "likes") {
    name
    likes
  }
  latestRecipes: recipes(orderBy: "createdAt") {
    name
    createdAt
  }
}
```

In this example `topRecipes` and `latestRecipes` are both a list of `Recipe`s but they are querying for different subsets. The alternative is to name them after the property but `createGetRecipesQueryMock_topRecipes` could be confusing since one could expect it to create an array. In order to avoid that confusion while staying close to graphql's semantics, it should probably be named `createGetRecipesQueryMock_topRecipesSelectionSet`. Here is what I think would be the desired output:

```typescript
import * as types from "./types";
import * as factories from "./factories";

export type GetRecipesQuery = /* type generated by typescript/operations */;

export function createGetRecipesQueryMock(props: Partial<GetRecipesQuery>): GetRecipesQuery {
  return {
    __typename: "Query",
    topRecipes: [],
    latestRecipes: [],
  };
}

export function createGetRecipesQueryMock_topRecipesSelectionSet(
  props: Partial<GetRecipesQuery["topRecipes"][number]>
): GetRecipesQuery["topRecipes"][number] {
  const {
    __typename,
    name,
    likes,
  } = factories.createRecipeMock(props);
  
  return {
    __typename,
    name,
    likes,
  };
}

export function createGetRecipesQueryMock_latestRecipesSelectionSet(
  props: Partial<GetRecipesQuery["latestRecipes"][number]>
): GetRecipesQuery["latestRecipes"][number] {
  const {
    __typename,
    name,
    createdAt,
  } = factories.createRecipeMock(props);
  
  return {
    __typename,
    name,
    createdAt,
  };
}
```

For nested fields, e.g `topRecipes.author`, it would be `createGetRecipesQueryMock_topRecipes_authorSelectionSet`.

**Comment by zhouzi** [comment]
Once the feature described in this issue is implemented, we should be good to publish a v1.0.0.

**Comment by zhouzi** [comment]
This feature is now implemented and available in `graphql-codegen-factories@1.0.0-beta.4` 🎉 Have a look at the [example](https://stackblitz.com/github/zhouzi/graphql-codegen-factories/tree/main/examples/usage-with-near-operation-file-preset) to see what it generates.

I will make a few tweaks to the examples, test some more and publish a `graphql-codegen-factories@1.0.0`.

**Comment by ertrzyiks** [comment]
That's great news @zhouzi ! I will play with play with beta.4 over the weekend (unless we have proper 1.0.0 already by then)
